tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9760,198 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+			log.FailOnError(err, "failed to get pool drive limit")
+			maxDriveLimitInt := int(maxDriveLimit)
+			maxDriveAllowed := maxDriveLimitInt - drvNumInt
+			err = Inst().V.RefreshDriverEndpoints()
+			log.FailOnError(err, "error refreshing volume endpoints")
+			//expand by adding disk
+			selectedNodeForOps, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "failed while getting node")
+			poolListForOps, err := GetPoolsDetailsOnNode(selectedNodeForOps)
+			log.FailOnError(err, "failed while getting updated node pool list")
+
+			//add max drive allowed per node
+			for maxDriveAllowed > 0 {
+				for i := 0; i < len(poolListForOps); i++ {
+					err = Inst().V.RefreshDriverEndpoints()
+					log.FailOnError(err, "error refreshing volume endpoints")
+					drvSize, err := getPoolDiskSize(poolListForOps[i])
+					log.FailOnError(err, "error getting drive size for pool [%s]", poolListForOps[i].Uuid)
+					drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+					log.FailOnError(err, "error in getting pool with IO with error")
+					if drvs, ok := drvMap[fmt.Sprintf("%d", poolListForOps[i].ID)]; ok {
+						if (POOL_MAX_CLOUD_DRIVES - len(drvs)) > 0 {
+							drivesThatCanBeAdded := (POOL_MAX_CLOUD_DRIVES - len(drvs))
+							for j := 1; j <= drivesThatCanBeAdded; j++ {
+								driveSize := drvSize * uint64(j)
+								expectedSize := (poolListForOps[i].TotalSize / units.GiB) + driveSize
+								isjournal, err := isJournalEnabled()
+								log.FailOnError(err, "Failed to check is journal enabled")
+								expectedSizeWithJournal := expectedSize
+								if isjournal {
+									journalSize := uint64(3)
+									expectedSizeWithJournal = expectedSizeWithJournal - journalSize
+								}
+								err = Inst().V.ExpandPool(poolListForOps[i].Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+								log.FailOnError(err, "error while expanding pool")
+								maxDriveAllowed = maxDriveAllowed - 1
+								if maxDriveAllowed == 0 {
+									break
+								}
+								resizeErr := waitForPoolToBeResized(expectedSize, poolListForOps[i].Uuid, isjournal)
+								dash.VerifyFatal(resizeErr, nil, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+								err = Inst().V.RefreshDriverEndpoints()
+								log.FailOnError(err, "error refreshing volume endpoints")
+							}
+						}
+					}
+				}
+			}
+			//get a pool where we can add one more drive
+			sNode, err := node.GetNodeByName(selectedNode.Name)
+			log.FailOnError(err, "failed while getting updated node pool list")
+			poolListOnNode, err := GetPoolsDetailsOnNode(sNode)
+			for i := 0; i < len(poolListOnNode); i++ {
+				drvMap, err := Inst().V.GetPoolDrives(&sNode)
+				log.FailOnError(err, "error getting pool drives from node [%s]", sNode.Name)
+				if drvs, ok := drvMap[fmt.Sprintf("%d", poolListOnNode[i].ID)]; ok {
+					if (POOL_MAX_CLOUD_DRIVES - len(drvs)) > 0 {
+						selectedPools = poolListOnNode[i].Uuid
```

Comment:
can you explain this? if you are already adding max allowed drives in earlier workflow, this condition will never be satisfied and test fails here

Reply:
this is for other pool not the one with max drives in previous case